xfrd-tcp.c

```diff
@@ -775,6 +992,104 @@ xfrd_tcp_write(struct xfrd_tcp_pipeline* tp, xfrd_zone_type* zone)
 	tcp_pipe_reset_timeout(tp);
 }
 
+#ifdef HAVE_TLS_1_3
+static int
+conn_read_ssl(struct xfrd_tcp* tcp, SSL* ssl)
+{
+	ssize_t received;
+	/* receive leading packet length bytes */
+	if(tcp->total_bytes < sizeof(tcp->msglen)) {
+		ERR_clear_error();
+		received = SSL_read(ssl,
+						(char*) &tcp->msglen + tcp->total_bytes,
+						sizeof(tcp->msglen) - tcp->total_bytes);
+		if (received <= 0) {
+			int err = SSL_get_error(ssl, received);
+			log_msg(LOG_ERR, "ssl_read returned error %d", err);
+			if(err == SSL_ERROR_ZERO_RETURN) {
+				/* EOF */
+				return 0;
+			}
+		}
+		if(received == -1) {
+			if(errno == EAGAIN || errno == EINTR) {
+				/* read would block, try later */
+				return 0;
+			} else {
+#ifdef ECONNRESET
+				if (verbosity >= 2 || errno != ECONNRESET)
+#endif /* ECONNRESET */
+					log_msg(LOG_ERR, "tls read sz: %s", strerror(errno));
+				return -1;
+			}
+		} else if(received == 0) {
+			/* EOF */
+			return -1;
+		}
+		tcp->total_bytes += received;
+		if(tcp->total_bytes < sizeof(tcp->msglen)) {
+			/* not complete yet, try later */
+			return 0;
+		}
+
+		assert(tcp->total_bytes == sizeof(tcp->msglen));
+		tcp->msglen = ntohs(tcp->msglen);
+
+		if(tcp->msglen == 0) {
+			buffer_set_limit(tcp->packet, tcp->msglen);
+			return 1;
+		}
+		if(tcp->msglen > buffer_capacity(tcp->packet)) {
+			log_msg(LOG_ERR, "buffer too small, dropping connection");
+			return 0;
+		}
+		buffer_set_limit(tcp->packet, tcp->msglen);
+	}
+
+	assert(buffer_remaining(tcp->packet) > 0);
+	ERR_clear_error();
+
+	received = SSL_read(ssl, buffer_current(tcp->packet),
+					buffer_remaining(tcp->packet));
+
+	if (received <= 0) {
+		int err =SSL_get_error(ssl, received);
```

Comment:
Based on other code, you want this space.

```suggestion
		int err = SSL_get_error(ssl, received);
```

Reply:
I saw the space in the current code, so resolved.